siliconcompiler/tools/openroad/openroad.py

```diff
@@ -398,38 +395,39 @@ def get_metric_sources(metric):
 
 
 ######
-def get_pex_corners(chip):
-
+def get_library_timing_keypaths(chip, lib):
     step = chip.get('arg', 'step')
     index = chip.get('arg', 'index')
 
-    corners = set()
+    delaymodel = chip.get('asic', 'delaymodel', step=step, index=index)
+    keypaths = {}
     for constraint in chip.getkeys('constraint', 'timing'):
-        pexcorner = chip.get('constraint', 'timing', constraint, 'pexcorner',
-                             step=step, index=index)
-        if pexcorner:
-            corners.add(pexcorner)
+        corners = chip.get('constraint', 'timing', constraint, 'libcorner', step=step, index=index)
+        for corner in corners:
+            if chip.valid('library', lib, 'output', corner, delaymodel):
+                keypaths[constraint] = ('library', lib, 'output', corner, delaymodel)
```

Comment:
Are the corners sorted in any way?
The last overwrite is the last/most ??? corner that is valid?

Reply:
not sorted, this is just ensuring that the liberty models are marked as required